src/GeositeFramework/plugins/layer_selector_v2/main.js

```diff
@@ -160,14 +179,47 @@ define([
                     }
                 }, this);
 
-                _.each(visibleLayerIds, function(layerIds, serviceUrl) {
+                _.each(visibleLayerIds, function(layerServiceIds, serviceUrl) {
                     var mapLayer = this.map.getLayer(serviceUrl);
-                    if (layerIds.length === 0) {
+                    if (layerServiceIds.length === 0) {
                         mapLayer.setVisibleLayers([-1]);
                     } else {
-                        mapLayer.setVisibleLayers(layerIds);
+                        mapLayer.setVisibleLayers(layerServiceIds);
                     }
                 }, this);
+
+                this._setOpacityForService(visibleLayerIds);
+            },
+
+            _setOpacityForService: function(visibleLayerIds) {
+                // VisibleLayerIds is a map of serviceUrls -> selected layer service Ids
+                _.each(visibleLayerIds, function(layerServiceIds, serviceUrl) {
+                    var mapLayer = this.map.getLayer(serviceUrl),
+                        self = this;
+
+                    if (layerServiceIds.length !== 0) {
+                        var optionsArray = [],
+                            selectedLayers = this.state.getSelectedLayersForService(serviceUrl);
```

Comment:
Oh, I see. Instead of iterating over `visibleLayerIds` you might just want to create another loop based on this code:  https://github.com/caseypt/GeositeFramework/blob/feature/opacity-slider-2/src/GeositeFramework/plugins/layer_selector_v2/main.js#L163.

Another alternative might be something like this?

```
layerByService = groupBy(getSelectedLayers(), layer => layer.getServiceUrl())
getDrawingOpts = layers => reduce(...)
each(layerByService, (layers, serviceUrl) => {
    drawingOpts = getDrawingOpts(layers)
    mapLayer = map.getLayer(serviceUrl)
    mapLayer.setLayerDrawingOptions(drawingOpts)
})

```